### PR TITLE
Multi year docs

### LINKED ISSUE #819: Update documentation - formulation (note that formulation needs updates not only in the investments, but also in ramping, groups, and binary charging)

### PR DESCRIPTION
**Pull request by gnawin**
## Related issues

Closes #819 

## Checklist

- [x] I am following the [contributing guidelines](https://github.com/TulipaEnergy/TulipaEnergyModel.jl/blob/main/docs/src/90-contributing.md)

- [x] Tests are passing
- [x] Lint workflow is passing
- [ ] Docs were updated and workflow is passing

### CODE REVIEW

File: docs/src/40-formulation.md [hunk 1]
```diff
@@ -196,54 +336,69 @@ unit\_on\_cost &= \sum_{a \in \mathcal{A}^{\text{uc}}} \sum_{k \in \mathcal{K}}
...
+\\ \\ \forall y \in \mathcal{Y}, \forall a \in \mathcal{A}^{\text{i}}_y \cap \mathcal{A}^{\text{sb}}_y, \forall k_y \in \mathcal{K}_y,\forall b_{k_y} \in \mathcal{B_{k_y}}
+\end{aligned}
+```
 
 ```math
 \begin{aligned}
-\sum_{f \in \mathcal{F}^{\text{out}}_a} v^{\text{flow}}_{f,k,b_k} \leq p^{\text{availability profile}}_{a,k,b_k} \cdot \left(p^{\text{capacity}}_{a} \cdot p^{\text{init units}}_{a} + p^{\text{inv limit}}_{a} \right) \cdot \left(1 - v^{\text{is charging}}_{a,k,b_k} \right) \quad
-\\ \\ \forall a \in \mathcal{A}^{\text{sb}}, \forall k \in \mathcal{K},\forall b_k \in \mathcal{B_k}
+\sum_{f \in \mathcal{F}^{\text{out}}_{a,y}} v^{\text{flow}}_{f,k_y,b_{k_y}} \leq p^{\text{availability profile}}_{a,y,k_y,b_{k_y}} \cdot p^{\text{capacity}}_{a} \cdot p^{\text{init units}}_{a,y} \cdot \left(1 - v^{\text{is charging}}_{a,k_y,b_{k_y}} \right) \quad
+\\ \\ \forall y \in \mathcal{Y}, \forall a \in , \forall k_y \in \mathcal{K}_y,\forall b_{k_y} \in \mathcal{B_{k_y}}
```
Comment (gnawin) [inline docs/src/40-formulation.md:358]: What is the set for a?

File: docs/src/40-formulation.md [hunk 2]
```diff
@@ -196,54 +336,69 @@ unit\_on\_cost &= \sum_{a \in \mathcal{A}^{\text{uc}}} \sum_{k \in \mathcal{K}}
...
 \end{aligned}
 ```
 
 ```math
 \begin{aligned}
-\sum_{f \in \mathcal{F}^{\text{in}}_a} v^{\text{flow}}_{f,k,b_k} \leq p^{\text{availability profile}}_{a,k,b_k} \cdot p^{\text{capacity}}_{a} \cdot \left(p^{\text{init units}}_{a} \cdot v^{\text{is charging}}_{a,k,b_k} + v^{\text{inv}}_{a} \right)   \quad \forall a \in \mathcal{A}^{\text{sb}}, \forall k \in \mathcal{K},\forall b_k \in \mathcal{B_k}
+\sum_{f \in \mathcal{F}^{\text{in}}_{a,y}} v^{\text{flow}}_{f,k_y,b_{k_y}} \leq p^{\text{availability profile}}_{a,y,k_y,b_{k_y}} \cdot p^{\text{capacity}}_{a} \cdot p^{\text{init units}}_{a,y}  \cdot v^{\text{is charging}}_{a,k_y,b_{k_y}} \quad \forall y \in \mathcal{Y}, \forall a \in , \forall k_y \in \mathcal{K_y},\forall b_{k_y} \in \mathcal{B_{k_y}}
```
Comment (gnawin) [inline docs/src/40-formulation.md:388]: What is the set for a?

**Comment by gnawin** [comment]
@datejada This PR is finished from my side. It would be nice if you can have a look, at least to see if there are things you can spot roughly, and even better if you could read it more carefully. The branch is in main so you can pull and view locally.

In particular, I'm not sure about the storage part: storage binary and storage constraints. I tried my best to document what's in the code, but since you coded this part, it is probably easier for you to check carefully (most probably I made mistakes in interpreting your code). I also left two comments (see above), where something was missing. 

Thank you!

**Comment by datejada** [comment]
> @datejada This PR is finished from my side. It would be nice if you can have a look, at least to see if there are things you can spot roughly, and even better if you could read it more carefully. The branch is in main so you can pull and view locally.
> 
> In particular, I'm not sure about the storage part: storage binary and storage constraints. I tried my best to document what's in the code, but since you coded this part, it is probably easier for you to check carefully (most probably I made mistakes in interpreting your code). I also left two comments (see above), where something was missing.
> 
> Thank you!
> 
> And feel free to merge it afterward.
> 
> Ps. good to mention that this PR is only about formulation, I will start another one on "How-to".

Thanks! I have updated the sets you commented on. If you have doubts, please let me know.

**Review by datejada — APPROVED** [review]

**Pull request merged**